types/index.d.ts

```diff
@@ -218,16 +239,20 @@ export interface SmartActionLoadHookField extends SmartActionHookField {
   position: number,
 }
 
+export interface SmartActionHookRequest extends ForestRequest {
+  body: SelectAllRequestBody,
+}
+
 export interface SmartActionLoadHook<M extends Sequelize.Model = any> {
-  (context: { fields: SmartActionLoadHookField[], record: M }): SmartActionLoadHookField[]
+  (context: { fields: SmartActionLoadHookField[], request: SmartActionHookRequest }): SmartActionLoadHookField[]
 }
 
 export interface SmartActionChangeHookField extends SmartActionHookField {
   previousValue: any,
 }
 
 export interface SmartActionChangeHook<M extends Sequelize.Model = any> {
-  (context: { fields: SmartActionChangeHookField[], record: M, changedField: SmartActionChangeHookField }): SmartActionChangeHookField[]
+  (context: { fields: SmartActionChangeHookField[], changedField: SmartActionChangeHookField, request: SmartActionHookRequest }): SmartActionChangeHookField[]
```

Comment:
Actually, the M generic type is not used anymore. If we don't provide any record to the user, you can remove it.